main.cpp

```diff
@@ -56,64 +56,81 @@ Create a branch named Part1
 #include <string>
 struct T
 {
-    T(<#type name#> v, const char* <#variable name#>)   //1
-    //2
-    //3
+    T(float v, const char* w) : video(v), waveForm(w) {}  //1
+    float video; //2
+    std::string waveForm; //3
 };
 
-struct <#structName1#>                                //4
+struct CompareCalc                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
+        if( a->video < b->video ) return a;
+        if( a->video > b->video ) return b;
```

Comment:
What's the one thing we always do with pointers before we use them? 

- [ ] FIXME